.github/workflows/_chromatic.yml

```
@@ -0,0 +1,50 @@
+name: "Publish to chromatic"
+on:
+  workflow_call:
+    inputs:
+      working-directory:
+        required: true
+        description: The directory where Storybook should be published from
+        type: string
+    secrets:
+      CHROMATIC_PROJECT_TOKEN:
+        required: true
+        description: Chromatic project token for publishing
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref
+    }}
+  cancel-in-progress: true
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+        # "prepare" hook on `bun install` runs `bun run build`
+      - name: Setup environment
+        uses: ./.github/actions/setup-env
+
+      - name: Code quality checks
+        run: bun run check
+
+      - name: Test
+        run: bun run test
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+        # "prepare" hook on `bun install` runs `bun run build`
+      - name: Setup environment
+        uses: ./.github/actions/setup-env
+
+      - name: Publish to Chromatic
+        uses: chromaui/action@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+          exitZeroOnChanges: true
+          workingDir: ${{ inputs.working-directory }}
+          buildScriptName: "build:storybook"
```

Chromatic [recommends](https://www.chromatic.com/docs/github-actions/#maintain-a-clean-main-branch) we consider the main branch to be "auto-accepted" - whatever is merged to main should be an approved visual diff.

With Percy, this would mean running a separate test on pushes to `main`, but Chromatic makes this a bit easier: all we need to do is use `autoAcceptChanges: "main".

```suggestion
          buildScriptName: "build:storybook"
          autoAcceptChanges: "main"
```